src/mechglue/gpp_misc.c

```diff
@@ -2,22 +2,20 @@
 
 #include "gss_plugin.h"
 
+/* NOTE: The mech_invoke() methods are intrinsically intended to only
+ * influence local process beahvior, or act on local process memory.
+ * Therefore there is no point in even trying to proxy it. Always bridge
+ * it back to the local mechanism. We'll handle specific desired_objects
+ * in future if any exist that make sense to be proxied */
 OM_uint32 gssi_mech_invoke(OM_uint32 *minor_status,
                            const gss_OID desired_mech,
                            const gss_OID desired_object,
                            gss_buffer_t value)
 {
-    enum gpp_behavior behavior;
     OM_uint32 maj, min;
 
     GSSI_TRACE();
 
-    /* FIXME: implement remote invoke mech, only local for now */
-    behavior = gpp_get_behavior();
-    if (behavior == GPP_REMOTE_ONLY) {
```

Comment:
The comment update I understand, but I'm confused why you removed this check.  Shouldn't it still be a failure?

Reply:
No, I fully consider this to be a local request even when REMOTE_ONLY is requested ... for example if you request a lucid context you need to be able to free it, regardless of the mode of operation, same for kdc context for kadm5

So in this light I consider failing on the REMOTE_ONLY case to be a bug.

Reply:
Okay, makes sense.